worlds/kdl3/__init__.py

```diff
@@ -300,67 +325,63 @@ def generate_basic(self) -> None:
         else:
             self.boss_butch_bosses = [False for _ in range(6)]
 
-    def generate_output(self, output_directory: str):
-        rom_path = ""
+    def generate_output(self, output_directory: str) -> None:
         try:
-            rom = RomData(get_base_rom_path())
-            patch_rom(self, rom)
+            patch = KDL3ProcedurePatch()
+            patch_rom(self, patch)
 
-            rom_path = os.path.join(output_directory, f"{self.multiworld.get_out_file_name_base(self.player)}.sfc")
-            rom.write_to_file(rom_path)
-            self.rom_name = rom.name
+            self.rom_name = patch.name
 
-            patch = KDL3DeltaPatch(os.path.splitext(rom_path)[0] + KDL3DeltaPatch.patch_file_ending, player=self.player,
-                                   player_name=self.multiworld.player_name[self.player], patched_path=rom_path)
-            patch.write()
+            patch.write(os.path.join(output_directory,
+                                     f"{self.multiworld.get_out_file_name_base(self.player)}{patch.patch_file_ending}"))
         except Exception:
             raise
         finally:
             self.rom_name_available_event.set()  # make sure threading continues and errors are collected
-            if os.path.exists(rom_path):
-                os.unlink(rom_path)
 
-    def modify_multidata(self, multidata: dict):
+    def modify_multidata(self, multidata: Dict[str, Any]) -> None:
         # wait for self.rom_name to be available.
         self.rom_name_available_event.wait()
+        assert isinstance(self.rom_name, bytes)
         rom_name = getattr(self, "rom_name", None)
         # we skip in case of error, so that the original error in the output thread is the one that gets raised
         if rom_name:
-            new_name = base64.b64encode(bytes(self.rom_name)).decode()
+            new_name = base64.b64encode(self.rom_name).decode()
             multidata["connect_names"][new_name] = multidata["connect_names"][self.multiworld.player_name[self.player]]
 
     def write_spoiler(self, spoiler_handle: TextIO) -> None:
         if self.stage_shuffle_enabled:
             spoiler_handle.write(f"\nLevel Layout ({self.multiworld.get_player_name(self.player)}):\n")
-            for level in LocationName.level_names:
-                for stage, i in zip(self.player_levels[LocationName.level_names[level]], range(1, 7)):
+            for level in location_name.level_names:
+                for stage, i in zip(self.player_levels[location_name.level_names[level]], range(1, 7)):
                     spoiler_handle.write(f"{level} {i}: {location_table[stage].replace(' - Complete', '')}\n")
         if self.options.animal_randomization:
             spoiler_handle.write(f"\nAnimal Friends ({self.multiworld.get_player_name(self.player)}):\n")
-            for level in self.player_levels:
+            for lvl in self.player_levels:
```

Comment:
What was the point of renaming this to a less explicit name?

Reply:
mypy would throw an error about the typing of `level` since it is used earlier for a prior loop.